### PR TITLE
enh(openid): Implement client_secret_basic as token_endpoint_auth_method

### PR DESCRIPTION
**Pull request by mayrstefan**
## Description

Centreon assumes that it can use client_secret_post as token_endpoint_auth_method. This seems to work for Keycloak as IdP but does not strictly follow the spec. All systems have to implement client_secret_basic. When a client is registered and no method ist set, the default is client_secret_basic. Other methods are optional.

**Fixes**
We have an access management solution that is very strict an we currently require the client_secret_basic method to be used

#9959

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [x] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [x] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.04.x
- [ ] 20.10.x
- [x] 21.04.x
- [x] 21.10.x (master)

<h2> How this pull request can be tested ? </h2>

All current OpenID Connect tests should work with both methods.

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Comment by lpinsivy** [comment]
internal ticket: MON-10783

File: www/class/centreonAuth.SSO.class.php [hunk 1]
```diff
@@ -365,23 +369,28 @@ public function getOpenIdConnectToken(
         string $clientId,
         string $clientSecret,
         string $code,
+        bool $clientBasicAuth,
```
Comment (lpinsivy) [inline www/class/centreonAuth.SSO.class.php:372]: can you also modify function definition?

File: www/class/centreonAuth.SSO.class.php [hunk 2]
```diff
@@ -527,24 +537,29 @@ public function refreshToken(
         string $clientId,
         string $clientSecret,
         string $refreshToken,
+        bool $clientBasicAuth,
```
Comment (lpinsivy) [inline www/class/centreonAuth.SSO.class.php:540]: can you also modify function definition?

File: www/class/centreonAuth.SSO.class.php [hunk 3]
```diff
@@ -586,21 +601,26 @@ public function logout(
...
```
Comment (lpinsivy) [inline www/class/centreonAuth.SSO.class.php:604]: can you also modify function definition?

**Review by lpinsivy — CHANGES_REQUESTED** [review]

File: www/class/centreonAuth.SSO.class.php (hunk 1 above)
Comment (mayrstefan) [reply www/class/centreonAuth.SSO.class.php:373]: Sorry, added the function documentation for this parameter

File: www/class/centreonAuth.SSO.class.php (hunk 2 above)
Comment (mayrstefan) [reply www/class/centreonAuth.SSO.class.php:542]: Sorry, added the function documentation for this parameter

File: www/class/centreonAuth.SSO.class.php (hunk 3 above)
Comment (mayrstefan) [reply www/class/centreonAuth.SSO.class.php:607]: Sorry, added the function documentation for this parameter

**Review by jeremyjaouen — APPROVED** [review]

**Review by lpinsivy — APPROVED** [review]

**Pull request merged**